homeassistant/components/stream/hls.py

```diff
@@ -64,55 +78,184 @@ class HlsPlaylistView(StreamView):
     cors_allowed = True
 
     @staticmethod
-    def render_preamble(track):
-        """Render preamble."""
-        return [
-            "#EXT-X-VERSION:7",
-            f"#EXT-X-TARGETDURATION:{track.target_duration}",
-            '#EXT-X-MAP:URI="init.mp4"',
-        ]
+    def render(track, ll_hls):
+        """Render HLS playlist file."""
+        # NUM_PLAYLIST_SEGMENTS+1 because most recent is probably not yet complete
+        segments = list(track.get_segment())[-(NUM_PLAYLIST_SEGMENTS + 1) :]
 
-    @staticmethod
-    def render_playlist(track):
-        """Render playlist."""
-        segments = list(track.get_segment())[-NUM_PLAYLIST_SEGMENTS:]
+        # Create a copy of the last segment with the parts frozen for rendering playlist
+        segments[-1] = copy(segments[-1])
+        segments[-1].parts = segments[-1].parts.copy()
```

Comment:
Copy the parts of the incomplete segment so they don't change when we iterate through below.